toolchain/lower/handle_call.cpp

```diff
@@ -318,6 +318,45 @@ static auto HandleBuiltinCall(FunctionContext& context, SemIR::InstId inst_id,
       return;
     }
 
+    case SemIR::BuiltinFunctionKind::StringAt: {
+      // Get the string argument  
+      auto string_inst_id = arg_ids[0];
+      auto* string_arg = context.GetValue(string_inst_id);
+      
+      // Check if this is a pointer to a String or a String value
+      llvm::Value* string_value;
+      if (string_arg->getType()->isPointerTy()) {
+        auto string_type_id = context.GetTypeIdOfInst(string_inst_id);
+        auto* string_type = context.GetType(string_type_id);
+        string_value = context.builder().CreateLoad(
+            string_type, string_arg, "string.load");
+      } else {
+        // Already have the struct value
+        string_value = string_arg;
+      }
+      
+      // String is a struct with ptr and size fields
+      auto* string_ptr_field = context.builder().CreateExtractValue(
+          string_value, {0}, "string.ptr");
+      
+      // Get the index value
+      auto* index_value = context.GetValue(arg_ids[1]);
+      
+      // Compute the address of the character at the given index
+      auto* char_ptr = context.builder().CreateInBoundsGEP(
+          llvm::Type::getInt8Ty(context.llvm_context()),
+          string_ptr_field, index_value, "string.char_ptr");
+      
+      auto* char_i8 = context.builder().CreateLoad(
+          llvm::Type::getInt8Ty(context.llvm_context()),
+          char_ptr, "string.char");
+      
+      // Extend to i32 
+      context.SetLocal(inst_id, context.builder().CreateZExt(
+          char_i8, context.GetTypeOfInst(inst_id), "string.char.zext"));
```

Comment:
Is this extending to i32? Isn't it going to whatever the return type was? Maybe drop the comment?